### PR TITLE
Perspective transition

### PR DESCRIPTION
**Pull request by drhaynes**
Adds antialiased 3d rotated transition to the container view.

screen shot 2015-03-18 at 14 49 18

Still to do (will be in a subsequent PR):
- full pan gesture interactivity.
- a few minor issues to sort, mostly involving the right hand menu.

### CODE REVIEW

File: RESideMenu/RESideMenuController.m [hunk 1]
```diff
@@ -367,18 +429,39 @@ - (void)showLeftMenuViewController
         [self.leftMenuViewController endAppearanceTransition];
         [self.contentViewController endAppearanceTransition];
     }];
-    
+
+    // Perspective rotation animation
+    CABasicAnimation *animation = [CABasicAnimation animationWithKeyPath:@"transform.rotation.y"];
+    animation.fromValue = @0.0;
+    animation.toValue = [NSNumber numberWithDouble:self.perspectiveRotationAmountRadians];
```
Comment (dhardiman) [inline RESideMenu/RESideMenuController.m]: `@(x)` instead of `[NSNumber numberWithFloat:x]`?

File: RESideMenu/RESideMenuController.m [hunk 2]
```diff
@@ -412,29 +495,45 @@ - (void)showRightMenuViewController
         [self.rightMenuViewController endAppearanceTransition];
         [self.contentViewController endAppearanceTransition];
     }];
-    
+
+    // Perspective rotation animation
+    CABasicAnimation *animation = [CABasicAnimation animationWithKeyPath:@"transform.rotation.y"];
```
Comment (dhardiman) [inline RESideMenu/RESideMenuController.m]: Worth extracting these animation creations out to a method? These two share a lot in common with the two above.

File: RESideMenu/RESideMenuController.m (hunk 1 above)
Comment (drhaynes) [inline RESideMenu/RESideMenuController.m]: Thanks for the reminder!

File: RESideMenu/RESideMenuController.m (hunk 2 above)
Comment (drhaynes) [inline RESideMenu/RESideMenuController.m]: Yes definitely, will update.

**Comment by drhaynes** [comment]
Ready for review again.

**Pull request merged**